lib/puppet/type/ec2_instance.rb

```diff
@@ -22,9 +22,10 @@ def change_to_s(current, desired)
       current == desired ? current : "changed #{current} to #{desired}"
     end
     def insync?(is)
-      is = :present if is == :running
-      is = :stopped if is == :stopping
-      is.to_s == should.to_s
+      is.to_s == should.to_s or
+        (is.to_s == 'running' and should.to_s == 'present' ) or
+        (is.to_s == 'stopped' and should.to_s == 'present' ) or
+        (is.to_s == 'stopping' and should.to_s == 'stopped' )
```

Comment:
Assuming this is transitive, the last two lines would imply that `is.to_s == 'stopping' and should.to_s == 'present'` also needs to be added.